plugins/chat/spec/system/uploads_spec.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+describe "Uploading files in chat messages", type: :system, js: true do
+  fab!(:current_user) { Fabricate(:user) }
+  fab!(:channel_1) { Fabricate(:chat_channel) }
+  fab!(:message_1) { Fabricate(:chat_message, chat_channel: channel_1) }
+
+  let(:chat) { PageObjects::Pages::Chat.new }
+  let(:channel) { PageObjects::Pages::ChatChannel.new }
+
+  before { chat_system_bootstrap }
+
+  context "when uploading to a new message" do
+    before do
+      channel_1.add(current_user)
+      sign_in(current_user)
+    end
+
+    it "allows uploading a single file" do
```

Comment:
I could add a multi-file upload spec too, that may be useful, I am not sure if there is a way to do this with `attach_file` or if I just have to call it a few times